src/main/scala/fi/oph/koski/raportit/PerusopetuksenRaportitRepository.scala

```diff
@@ -196,8 +202,75 @@ case class PerusopetuksenRaportitRepository(db: DB) extends KoskiDatabaseMethods
       group by
         oo.opiskeluoikeus_oid"""
   }
+
+  // Tarkoituksena on hakea raportille tapauksia jotka ovat potentiaalisesti keskeyttäneet opinnot, mutta tätä tietoa ei ole siirretty Koskeen.
+  private def haettuVuosiluokkaViimeisinVahvistettuSuoritus(oppilaitos: Set[Organisaatio.Oid], paiva: LocalDate, vuosiluokka: String): Seq[Tunnisteet] = {
+    aktiivistenSuoritukset(oppilaitos.toSeq, Date.valueOf(paiva), vuosiluokka)
+      .foldLeft[List[Tunnisteet]](Nil)(viimeisinSuoritusOnHaettuVuosiluokka(vuosiluokka))
```

Comment:
Näitä foldeja on mun mielestä vaikea lukea. Erityisesti jos se operaatio on siirretty jonnekin muualle. Joutuu hyppimään koodissa ylös alas. Mutta miksei vaan voisi käyttää helpommin luettavaa `map`-operaatiota? Esim:
```
    aktiivistenSuoritukset(oppilaitos.toSeq, Date.valueOf(paiva), vuosiluokka)
      .map { case (opiskeluoikeusOid, vuosiluokanSuoritukset, aikajaksoIds) =>
        val vahvistetutVuosiluokat = vuosiluokanSuoritukset
          .takeWhile { x => x.vuosiluokka == vuosiluokka && x.vahvistuspäivä.isDefined }
          .map(_.id)

        (opiskeluoikeusOid, vahvistetutVuosiluokat, aikajaksoIds)
      }.filter(_._2.nonEmpty)
```
Jos ei tykkää tuosta `filter`:istä niin sitten voisi harkita esim `flatMap`:in käyttöä